### ISSUE TITLE
[YAML] Add Comments to Yaml

### ISSUE THREAD

**Issue by Articdive** [TheElectronWill/Night-Config#57]
I'm aware of SnakeYaml not supporting comments during the parse, however since you have your own yaml writer you can definitely support outputting comments, you would just need to bridge the YamlCommentedConfiguration's comment Map to your Writer and then when writing line for line, check if it's a node, if it is a node, add said comment for said node. Obviously you also need to calculate things such as Whitespace e.t.c and it wouldn't be the easiest solution, but it would allow for Comment output on Yaml

**Comment by Articdive** [comment]
A proof of concept would be [here ](https://github.com/Articdive/EnumToYAML/blob/master/src/main/java/de/articdive/enum_to_yaml/yaml/file/YAMLConfiguration.java)in the save() method.

**Comment by TheElectronWill** [comment]
Actually my "yaml writer" only exists to unwrap `Config` to `Map<String, Object>`. It then gives everything to snakeyaml, which does the real writing.

Supporting comments in yaml would indeed be very nice, both in reading and writing. However that would require more work and isn't a top priority, I'll add it to the backlog.

**Comment by Articdive** [comment]
I created my own custom implementation of this, if you are interested I can also PR it, however It's not the most efficient code and I'm nearly positive it can be improved.

**Comment by TheElectronWill** [comment]
Yep I'm interested. I would have liked to be able to read and write comments in order to preserve the configs but just writing comments is better than nothing :smiley:

**Comment by Articdive** [comment]
https://github.com/Articdive/EnumToConfiguration/tree/master/src/main/java/de/articdive/enum_to_configuration/yaml

**Comment by Articdive** [comment]
> Yep I'm interested. I would have liked to be able to read _and_ write comments in order to preserve the configs but just writing comments is better than nothing 😃

https://github.com/Articdive/EnumToConfiguration/tree/master/src/main/java/de/articdive/enum_to_configuration/yaml

**Comment by ArcheCraft** [comment]
Is this still considered?

**Comment by TheElectronWill** [comment]
It is, but it won't be ready before several months because it requires some work.

**Comment by maurxce** [comment]
Any updates on this?

**Comment by TheElectronWill** [comment]
Not yet, I would have to write a new yaml parser :sweat_smile:
